ui/js/app.js

```diff
@@ -136,28 +132,80 @@
 		update();
 	};
 
-	var step = window.location.search.substring(1);
-	var migrationFile = 'migration' + step + '.json';
+	var cycleNext = function cycleNext() {
+		step = (step !== '') ? parseInt(step, 10) + 1 : 1;
 
-	d3.json(migrationFile, function(error, json) {
-		if (error) {
-			console.log('>>>>> Error loading json file: ' + migrationFile);
-		}
+		if (step > 9) step = 1;
 
-		console.log(json.version);
+		window.location.assign(buildUrl(true));
+	};
 
-		json.graph.forEach(function(element) {
-			if (element.node) addNode(element.node);
-		});
+	var updateArchitecture = function updateArchitecture(e) {
+		var $target = $(this);
+
+		architecture = $target
+			.find('option:selected')
+			.val();
+
+		window.location.assign(buildUrl(false));
+	};
+
+	var buildUrl = function buildUrl(includeStep) {
+		var arch = (architecture) ? '?arch=' + architecture : '?arch=migration';
+		var s = (step) ? '&step=' + step : '&step=0';
+
+		var url = window.location.origin +
+				window.location.pathname +
+				arch ;
+
+		url = (includeStep) ? url + s : url;
+		return url;
+	};
+
+	var init = function init() {
+		var queryParams = window.location.search
+			.substring(1)
+			.split('&');
 
-		json.graph.forEach(function(element) {
-			if (element.edge) addEdge(element.source, element.target);
+		_(queryParams).each(function(param) {
+			var parts = param.split('=');
+			query[parts[0]] = parts[1];
 		});
 
-		update();
-	});
+		step = (query.step && query.step !== 0) ? query.step : '';
+		architecture = (query.arch) ? query.arch : 'migration';
+		var jsonFile = architecture + step + '.json';
+
+		d3.json(jsonFile, function(error, json) {
+			if (error) {
+				console.log('>>>>> Error loading json file: ' + jsonFile);
+				return;
+			}
+
+			console.log('>>>>> JSON version: ' + json.version);
+
+			$('#architecture').val(architecture);
+
+			if (architecture === 'lamp') $('#next').hide();
```

Comment:
We can assume that any architecture can have steps, Lamp doesn't right now.

My hack to reload the page for each step should just clear out the old data and reload the new file, but I didn't know how to do that.

Eventually, need to be able to add and remove individual nodes and edges dynamically.